### PR TITLE
fix: mutate loan reservation on mounted

### PR DESCRIPTION
**Pull request by christian14b**
Calling `updateLoanReservationTeam` when `TeamAttribution` is mounted to assign the loan to a team and also removing `updateLoanReservation` condition because when you select the same option the method is not triggered by `@update:modelValue`, it's being called when you select a different option.

### CODE REVIEW

File: src/components/Checkout/TeamAttribution.vue [hunk 1]
```diff
@@ -74,41 +74,40 @@ export default {
 	},
 	mounted() {
 		this.selectedId = this.teamId || 0;
+		this.updateLoanReservation();
```
Comment (dyersituations) [inline src/components/Checkout/TeamAttribution.vue:77]: Is the team ID definitely populated at this point?

File: src/components/Checkout/TeamAttribution.vue [hunk 2]
```diff
@@ -74,41 +74,40 @@ export default {
...
 	},
 	methods: {
 		updateLoanReservation() {
-			if (this.selectedId !== this.teamId) {
```
Comment (dyersituations) [inline src/components/Checkout/TeamAttribution.vue:80]: If we wait for the team ID to be defined in mounted, do we need to remove this check?

Comment (dyersituations) [reply src/components/Checkout/TeamAttribution.vue:80]: Or is this just in case?

**Comment by dyersituations** [comment]
@christian14b Seeing an interesting/confusing unit test failure.

**Comment by roger-in-kiva** [comment]
@dyersituations we are struggling with the thanks page locally, I think is monolity API, are you able to test this PR?

File: src/components/Checkout/TeamAttribution.vue (hunk 2 above)
Comment (dyersituations) [reply src/components/Checkout/TeamAttribution.vue:80]: Sounds like just in case, which I think is fine.

**Review by dyersituations — APPROVED** [review]

**Review by roger-in-kiva — APPROVED** [review]

**Review by dyersituations — APPROVED** [review]

**Pull request merged**